### PR TITLE
Shorten otel.instrumentation_library.* attribute names

### PR DESCRIPTION
**Pull request by tigrannajaryan**
otel.instrumentation_library.name and otel.instrumentation_library.version
are quite long. The keys increase the size of the wire data (and they are present
for every span emitted by OpenTelemetry).

Long keys may also cause trouble when formatting in the UI.

This change uses a shorter "otel.library" prefix. I think this does not create
any ambiguity, it appears that this is the only "library" OpenTelemetry-emitted
span should refer to.

I am not attached to any particular name, open to any other naming suggestions
as long as they are reasonably short.

### CODE REVIEW

**Comment by iNikem** [comment]
The same change should be done in Jaeger translation, right?

**Comment by tigrannajaryan** [comment]
> The same change should be done in Jaeger translation, right?

Correct. Updated the PR to include Jaeger.

**Review by bogdandrutu — APPROVED** [review]

**Review by jkwatson — APPROVED** [review]

**Review by iNikem — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]

**Review by arminru — COMMENTED** [review]
I think it could be mistaken for both the instrument*ed* library and even the OpenTelemetry SDK ("OTel library"), so I'd rename it to `otel.instrumentation` rather than `otel.library`.
If ambiguity is not an issue because we expect all consumers to pay close attention to the spec, the option with fewer character should work as well, however.

**Comment by anuraaga** [comment]
Hey - I filed #991 because if this will be locked in with GA, I think we're still missing a good story for non-OTLP that needs to be resolved and it would make the bytes of `instrumentation` vs `library` significant. I approved this PR because I think less data is guaranteed to be better than more, and I would generally go with the smaller word regardless of potential collision because of the cost. If we're not locked in we can still work on this though.

**Comment by arminru** [comment]
@anuraaga Neither blocking nor approving this PR, I just wanted to mention my concerns and provide an alternative but, as I said above, if ambiguity is not an issue for us, the shorter one will work as well.

**Comment by tigrannajaryan** [comment]
> I think it could be mistaken for both the instrument_ed_ library and even the OpenTelemetry SDK ("OTel library"), so I'd rename it to `otel.instrumentation` rather than `otel.library`.
> If ambiguity is not an issue because we expect all consumers to pay close attention to the spec, the option with fewer character should work as well, however.

I am fine if we go with a different name. I am not sure `otel.instrumentation.name` is clearer than `otel.library.name`. Absent other arguments I'd go with we the shorter version.

Let's keep this PR open just in case someone has a brilliant naming idea. Either way this needs to be merged or rejected by Friday when we freeze the trace spec.

**Comment by Oberon00** [comment]
I think `otel.library.name` is easily confused with the telemetry library name.

**Comment by arminru** [comment]
@tigrannajaryan Apart from the ambiguity, I'm not familiar with the serialization format and transport protocol used by Jaeger and Zipkin but personally I'd expect some compression to be applied on the transport layer. This would solve the problem with attribute keys (and also values) being duplicated on each individual span. Is this not the case here? I think that should rather be a concern of the format/protocol rather than trying to shorten keys (and values) in the payload. The value, which is usually a fully-qualified class name, will likely be longer than the key anyway and since I'd only expect very few different class names here in real world (1 to 20-ish in one trace/export), compressing them on the transport layer would achieve way better savings.

**Comment by tigrannajaryan** [comment]
> @tigrannajaryan Apart from the ambiguity, I'm not familiar with the serialization format and transport protocol used by Jaeger and Zipkin but personally I'd expect some compression to be applied on the transport layer. This would solve the problem with attribute keys (and also values) being duplicated on each individual span. Is this not the case here? I think that should rather be a concern of the format/protocol rather than trying to shorten keys (and values) in the payload.

Yes, typically gzipped. Wire size and processing time due to larger key names is still a concern when volumes are in TBs. Protocols indeed need to be concerned about compact representation and OTLP addresses this by making instrumentation library a field and not an attribute. Existing formats (like Zipkin and Jaeger) though are outside of our control, so we need to be mindful of how we use them.

Also, this does not address the UI formatting issues in any way.

In short: it is generally a good idea to try to keep attribute names reasonably short. IMO `otel.instrumentation_library.name` crosses the boundary of reasonable length.

**Comment by tigrannajaryan** [comment]
> I think `otel.library.name` is easily confused with the telemetry library name.

@Oberon00 are you referring to Otel SDK name, which by convention is recorded in `telemetry.sdk.name`?

Can you think of a better name that is shorter than `otel.instrumentation_library.name`? I'd be glad to see some ideas.

**Review by jmacd — APPROVED** [review]

**Comment by Oberon00** [comment]
In https://github.com/open-telemetry/opentelemetry-specification/pull/989#pullrequestreview-494471598, `otel.instrumentation` was already suggested.

**Comment by andrewhsu** [comment]
@tigrannajaryan is this an editorial change or is this required to be resolved soon (like today)?

**Comment by tigrannajaryan** [comment]
This should be resolved today.

I do not want to make last minute changes and restart the PR with `otel.instrumentation.*` (@Oberon00 I am fine with your proposal, just not enough time to gain consensus again).

@open-telemetry/technical-committee Because of lack of time I would like to suggest this: if any TC member objects the PR in its current form please block it, and do so before 3pm Pacific Time, otherwise I am going to merge it since formally it has enough approvals. If I get an objection I am going to close the PR without merging.

**Comment by carlosalberto** [comment]
LGTM

**Comment by carlosalberto** [comment]
Merging as nobody opposed it. Thanks for waiting @tigrannajaryan
